webrecorder/browsertrix
# Display exclusions & list of URLs in crawl queue

Add new crawl queue section to crawl detail view with exclusions table and queue list that updates every ~5 seconds.

Resolves https://github.com/webrecorder/browsertrix-cloud/issues/318.

### Manual testing
1. Update API base URL in `.env.local`: `API_BASE_URL=https://stanford.browsertrix.cloud`
2. Run app with `yarn start`
3. Create crawl template with exclusions using the Advanced config editor. Example:
```
exclude:
  - likes$
  - media$
  - with_replies$
  - followers$
  - following$
  - photos$
  - twitter.com/settings.*
  - skip-this-string
  - twitter\.com/match/this/text/
  ```
4. Start crawl and go to crawl detail page. Click "Exclusions" on side navigation. Verify exclusions & crawl queue is displayed as expected
5. Interact with pagination controls. Verify pages update as expected
6. Click on a URL. Verify page opens in new tab
7. Wait for crawl to finish or cancel/stop crawl. Verify "Crawl is not running." message is shown

### Screenshots
**New tab in crawl detail view:**
<img width="1015" alt="Screen Shot 2022-10-10 at 2 30 15 PM" src="https://user-images.githubusercontent.com/4672952/194955302-470bd5fc-df58-43ca-8df8-b799eb38eb47.png">

**Page in queue list:**
<img width="801" alt="Screen Shot 2022-10-10 at 2 30 23 PM" src="https://user-images.githubusercontent.com/4672952/194955326-378051f8-077c-49bf-844f-e4df19fe61a3.png">

**Tab when crawl is not running:**
<img width="1021" alt="Screen Shot 2022-10-10 at 2 35 14 PM" src="https://user-images.githubusercontent.com/4672952/194955987-0c89ddb0-5fbf-4d57-81b8-35fff4b1f1d7.png">


### TODO Follow-ups
- Tab name and headings will change when the exclusion editor is built
- Paginate exclusion table
- Typography & spacing pass to match mockups
